pkg/storage/writer.go

```diff
@@ -54,6 +55,65 @@ func (u *uploaderWriter) Close(ctx context.Context) error {
 	return u.uploader.CompleteUpload(ctx)
 }
 
+type uploaderCompressWriter struct {
+	*uploaderWriter
+	bytesWritten int
+	cap          int
+	gzipWriter   *gzip.Writer
+}
+
+func (u *uploaderCompressWriter) Write(ctx context.Context, p []byte) (int, error) {
```

Comment:
🤔 it is possible to make this function and `(*uploaderWriter).Write` share code? they look too similar.

Reply:
fixed